Dockerfile

```diff
@@ -39,24 +36,24 @@ RUN mkdir -p /tmp/protoc && \
 
 RUN GO111MODULE=off go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
 
+RUN chmod +x /usr/local/bin/* && \
+  upx --lzma /usr/local/bin/*
+
 WORKDIR tmp/commit0
 COPY . .
 RUN make build-deps && make build && \
-  mv commit0 /usr/local/bin
-RUN upx --lzma /usr/local/bin/*
+  mv commit0 /usr/local/bin && \
+  upx --lzma /usr/local/bin/commit0
```

Comment:
Separating this into two steps means most of the time we only need to re-compress the commit0 binary, not the other stuff that never changes.